File: src/main/java/main/rendering/Luminosity.java
package main.rendering;

import main.shape.Point;
import main.shape.Triangle;

import java.awt.*;
import java.util.List;
import static main.utils.VectorUtils.scalar;

public class Luminosity {
    public static void render(List<Triangle> triangles, Render render) {
        for (Triangle triangle : triangles) {
            Point n = triangle.normal();
            Point light = new Point(0, 0, -1);
            float intensity = scalar(n, light) / (n.length() * light.length());
           // System.out.println(intensity);
            if (intensity > 0)
                render.process(triangle, new Color((int) (intensity * 255), (int) (intensity * 255), (int) (intensity * 255)));
        }
    }
}

File: src/main/java/main/rendering/Luminosity.kt
package main.rendering

import main.shape.Point
import main.shape.Triangle
import main.shape.primtives.Face
import main.utils.VectorUtils.multiply

import main.utils.VectorUtils.scalar

class Luminosity {

    companion object {
        fun render(faces: List<Face>, render: Render, isLight: Boolean) {
            for (face in faces) {
                val triangle = Triangle(face.a.v, face.b.v, face.c.v)
                val light = Point(0.0, 0.3, 1.0)
                val intensity1 = if (isLight) scalar(face.a.vn.normalize(), light).toDouble() else 1.0
                val intensity2 = if (isLight) scalar(face.b.vn.normalize(), light).toDouble() else 1.0
                val intensity3 = if (isLight) scalar(face.c.vn.normalize(), light).toDouble() else 1.0
                render.process(triangle, face, intensity1, intensity2, intensity3)
            }
        }
    }
}